frontend/src/views/SecretMainPage/components/SecretListView/SecretDetailSidebar.tsx

```diff
@@ -412,9 +417,38 @@ export const SecretDetailSidebar = ({
                   )}
                 />
               </div>
-              <div className="ml-1 flex items-center space-x-2">
+              <div className="ml-1 flex items-center space-x-4">
+                <ProjectPermissionCan
+                  I={ProjectPermissionActions.Edit}
+                  a={subject(ProjectPermissionSub.Secrets, { environment, secretPath })}
+                >
+                  {(isAllowed) => (
+                    <Modal>
+                      <ModalTrigger asChild>
+                        <Button
+                          variant="outline_bg"
+                          className="w-full px-2 py-1"
+                          leftIcon={<FontAwesomeIcon icon={faRoad} />}
+                          isDisabled={!isAllowed}
```

Comment:
I think we should also check if this is even a secret that has references in it and disable this button if not, with a tooltip saying `This secret has no references` or something like that

Reply:
I don't think we really need to make it that much complex to be honest. Like its not like something compute heavy or anything like that.

Reply:
I just feel like it might be confusing to to see a reference tree button when no references are present?

Reply:
Ok then let's disable it

Reply:
or at least say no references in the dialog when it opens, just feels ambiguous